components/gantry/multiaxis/multiaxis.go

```diff
@@ -115,18 +116,28 @@ func (g *multiAxis) MoveToPosition(
 		return errors.Errorf("need position inputs for %v-axis gantry, have %v positions", len(g.subAxes), len(positions))
 	}
 
-	idx := 0
+	if len(positions) != len(g.lengthsMm) {
+		return errors.Errorf(
+			"number of input positions %v does not match total gantry axes length %v",
+			len(positions), len(g.lengthsMm),
+		)
+	}
+
+	jdx := 0
 	for _, subAx := range g.subAxes {
+
 		subAxNum, err := subAx.Lengths(ctx, extra)
 		if err != nil {
 			return err
 		}
 
-		err = subAx.MoveToPosition(ctx, positions[idx:idx+len(subAxNum)-1], worldState, extra)
-		if err != nil {
+		pos := positions[jdx : jdx+len(subAxNum)]
+		jdx += len(subAxNum)
+
+		err = subAx.MoveToPosition(ctx, pos, worldState, extra)
+		if err != nil && !errors.Is(err, context.Canceled) {
```

Comment:
@spieswl @nfranczak @raybjork  if you'd like an update. Also, made `CurrentInputs` and `GoToInputs` fallthrough functions rather than copy-pastes of the logic to MoveToPosition and Position

Reply:
OK, that matches our patterns in `arm` so I like the consistency.

Reply:
This is actually opposite from what we have in arm, where GoToInputs is the function that the basic function that is called by MoveToPosition.  Can you switch these to be consistent?